File: distros/J/JavonetPerlSdk/lib/Javonet/Binaries/Python/javonet/core/handler/ArrayGetRankHandler.py
from javonet.core.handler.AbstractCommandHandler import AbstractCommandHandler
#import numpy as np


class ArrayGetRankHandler(AbstractCommandHandler):
    def __init__(self):
        self._required_parameters_count = 1

    def process(self, command):
        try:
            if len(command.payload) != self._required_parameters_count:
                raise Exception("ArrayGetRankHandler parameters mismatch!")

            array = command.payload[0]
            rank = 0
            while(isinstance(array, list)):
                rank = rank + 1
                array = array[0]

            return rank

        except Exception as e:
            exc_type, exc_value = type(e), e
            new_exc = exc_type(exc_value).with_traceback(e.__traceback__)
            raise new_exc from None

File: distros/J/JavonetPerlSdk/lib/Javonet/Binaries/Ruby/Windows/X64/core/handler/array_get_rank_handler.rb
require_relative 'abstract_command_handler'

class ArrayGetRankHandler < AbstractCommandHandler
  def initialize
    @required_parameters_count = 1
  end
  def process(command)
    begin
      if command.payload.length < @required_parameters_count
        raise ArgumentError.new "Array get rank parameters mismatch"
      end
      array = command.payload[0]
      rank = 0
      while array.is_a? Array
        rank = rank + 1
        array = array[0]
      end
      return rank
    rescue Exception => e
      return e
    end
  end
end